src/components/Rockets.js

```diff
@@ -0,0 +1,67 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchData, reserveRocket, cancelRocket } from '../redux/features/rockets/RocketSlice';
+import styles from '../styles/Rocket.module.css';
+
+const RocketsList = () => {
+  const rockets = useSelector((state) => state.rocket.rockets);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchData());
+  }, [dispatch]);
+
+  const handleReserveClick = (id) => {
+    const newState = rockets.map((rocket) => {
+      if (rocket.id !== id) { return rocket; }
+      return { ...rocket, reserved: true };
+    });
+    dispatch(reserveRocket(newState));
```

Comment:
- [ ]  As previous reviewer suggested the two issue are still there 

-  "I am able to reserve a rocket but when I check a profile page and come back they are canceled itself address that"
- "Same as mission is being canceled without clicking the leave mission button please address that"

This means that I can not have a cancel button because while I navigate to another page, the reserves are cancelling themselves. Please check the requested changes very well so that your project can be approved